Fix Issue #141 part 2

aluu317 opened the pull request:
Use declarative widgets for this version of scotch demo. However, there's an issue with resetting table selection.

(Issue: Select a scotch X from the suggestion table, then pick a new scotch A from the drop down. Expected is that the chart should only show this new scotch A. But the bug is that the chart still shows both this new scotch A and the scotch X previously selected from table). This issue is being worked on <a href="https://github.com/jupyter-incubator/declarativewidgets/issues/163">here</a>.

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+    }
+   },
+   "source": [
+    "But since urth-viz-table needs the index (distillery) as a regular column with the data, we need to add that column."
```
"..., we need to reset the DataFrame index so that it becomes a column."

parente replied:
Same thing about the second mention of adding a column. We're not adding a column but rather turning the DataFrame index into a column.

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+    "\n",
+    "And because urth-viz-* requires the data series as columns as opposed to rows, we will have to transpose this dataframe.\n",
+    "\n",
+    "Then similarly to the similar table above, we also need to add the index (feature) as a regular column."
```
Same thing here: not adding a column but rather resetting the index as a column.

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+    }
+   },
+   "source": [
+    "We also need another function that takes a scotch name, a similar scotch name and returns a list of these two names. Scotch is the name user picks from the dropdown, and similar is one user picks from the suggestion table."
```
Why is this function being created? How are the widgets going to use it later? (I'm lost on my first read through.)

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+    }
+   },
+   "source": [
+    "## Dashboard Setup <span style=\"float: right; font-size: 0.5em\"><a href=\"#Got-Scotch?\">Top</a></span>"
```
Probably want to call this Widget Setup. Dashboard layout is waaayyyy at the bottom.

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+   "source": [
+    "Lastly, we will display the scotches features with a variety of urth-viz-* elements.\n",
+    "\n",
+    "We need to use urth-core-function to call *get_feature_keys*, give it the selected scotch name, the similar scotch name user clicked on, and the result of the function call is a list of these two names. We will assign the result to a variable called *feature_keys* that another function *get_features_for_chart* will make use of right after this."
```
OK. Here's where that mystery function comes into play. I'm starting to understand that it's used to build a tuple that'll be used to track the drop down selected scotch and the last clicked scotch.

parente commented on etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb:
```diff
@@ -0,0 +1,783 @@
+    }
+   },
+   "source": [
+    "Now, we define a template to allow user switch between different types of chart. We start with setting up an iron-selector element which includes the 3 icons for table, bar and pie chart. When user clicks on any of those 3 icons, the name of the chart is stored as variable *chartType*. Then we set up iron-pages with 3 children pages that contain the three types of chart, each chart type per  ...
```
I'm totally lost on what the pie chart is showing me. I expect a (violent) negative reaction from folks attuned to visualization about it too. Let's just stick with bar and table if those are the two that function properly and are reasonable for comparing properties of two items.

parente commented:
Good work. A few comments above, mostly about minor changes. Once those are into the notebook here, do you think we can merge this ahead of the widget fixes or should we hold off until the widget fixes are also delivered?

aluu317 commented:
I fixed those minor changes in the instructions, and removed the pie chart. If you're ok with merging now, we can definitely do that. Then once the fix for urth-viz-table comes in later, we'll be good!

parente commented:
:shipit:
